src/rpc/rawtransaction_util.cpp

```diff
@@ -154,6 +154,13 @@ CMutableTransaction ConstructTransaction(const UniValue& inputs_in, const UniVal
         rawTx.nLockTime = nLockTime;
     }
 
+    if (!version.isNull()) {
+        uint32_t nVersion = version.getInt<uint32_t>();
+        if (nVersion < TX_MIN_STANDARD_VERSION || nVersion > TX_MAX_STANDARD_VERSION)
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid parameter, version out of range(1~3)");
```

Comment:
This RPC error should be a formatted string so that it remains accurate as `TX_MIN_STANDARD_VERSION` and `TX_MAX_STANDARD_VERSION` get updated.